pkg/server/plugin/datastore/sql/sql_test.go

```diff
@@ -2045,5 +2047,2468 @@ func (s *PluginSuite) TestConfigure() {
+			paged:   "with-token",
+			query: `
+SELECT
+	E.id AS e_id,
+	E.entry_id AS entry_id,
+	E.spiffe_id,
+	E.parent_id,
+	E.ttl AS reg_ttl,
+	E.admin,
+	E.downstream,
+	E.expiry,
+	S.id AS selector_id,
+	S.type AS selector_type,
+	S.value AS selector_value,
+	B.trust_domain,
+	D.id AS dns_name_id,
+	D.value AS dns_name
+FROM
+	registered_entries E
+LEFT JOIN
+	(SELECT 1 AS joinItem UNION SELECT 2 UNION SELECT 3) AS joinItems ON TRUE
+LEFT JOIN
+	selectors S ON joinItem=1 AND E.id=S.registered_entry_id
+LEFT JOIN
+	dns_names D ON joinItem=2 AND E.id=D.registered_entry_id
+LEFT JOIN
+	(federated_registration_entries F INNER JOIN bundles B ON F.bundle_id=B.id) ON joinItem=3 AND E.id=F.registered_entry_id
+WHERE E.id IN (
+	SELECT id FROM (
+	SELECT id FROM (
+		SELECT id FROM registered_entries WHERE spiffe_id = ?
+	) listing WHERE id > ? ORDER BY id ASC LIMIT 1
+	) workaround_for_mysql_subquery_limit
+)
+ORDER BY e_id, selector_id, dns_name_id
+;`,
+		},
+	}
+
+	for _, testCase := range testCases {
+		name := testCase.dialect + "-list-"
+		if len(testCase.by) == 0 {
+			name += "all"
+		} else {
+			name += "by-" + strings.Join(testCase.by, "-")
+		}
+		if testCase.paged != "" {
+			name += "-paged-" + testCase.paged
+		}
+		t.Run(name, func(t *testing.T) {
+			req := new(datastore.ListRegistrationEntriesRequest)
+			switch testCase.paged {
+			case "":
+			case "no-token":
+				req.Pagination = &datastore.Pagination{
+					PageSize: 1,
+				}
+			case "with-token":
+				req.Pagination = &datastore.Pagination{
+					PageSize: 1,
+					Token:    "2",
+				}
+			default:
+				require.FailNow(t, "unsupported page case: %q", testCase.paged)
+			}
 
+			for _, by := range testCase.by {
+				switch by {
+				case "parent-id":
+					req.ByParentId = &wrappers.StringValue{
+						Value: "spiffe://parent",
+					}
+				case "spiffe-id":
+					req.BySpiffeId = &wrappers.StringValue{
+						Value: "spiffe://id",
+					}
+				case "selector-subset-one":
+					req.BySelectors = &datastore.BySelectors{
+						Selectors: []*common.Selector{{Type: "a", Value: "1"}},
+						Match:     datastore.BySelectors_MATCH_SUBSET,
+					}
+				case "selector-subset-many":
+					req.BySelectors = &datastore.BySelectors{
+						Selectors: []*common.Selector{{Type: "a", Value: "1"}, {Type: "b", Value: "2"}},
+						Match:     datastore.BySelectors_MATCH_SUBSET,
+					}
+				case "selector-exact-one":
+					req.BySelectors = &datastore.BySelectors{
+						Selectors: []*common.Selector{{Type: "a", Value: "1"}},
+						Match:     datastore.BySelectors_MATCH_EXACT,
+					}
+				case "selector-exact-many":
+					req.BySelectors = &datastore.BySelectors{
+						Selectors: []*common.Selector{{Type: "a", Value: "1"}, {Type: "b", Value: "2"}},
+						Match:     datastore.BySelectors_MATCH_EXACT,
+					}
+				default:
+					require.FailNow(t, "unsupported by case: %q", by)
+				}
+			}
+
+			query, _, err := buildListRegistrationEntriesQuery(testCase.dialect, req)
```

Comment:
I wonder if there is a library out there that can tell us if this is valid SQL. Will be tough since its dialect specific.